### PR TITLE
Feature/crpl 2757 b token deployment

### PR DESCRIPTION
**Pull request by gyhevizi**

### CODE REVIEW

File: blockchain/ethereum/contracts/erc721/presets/erc721presetminterpauserautoid deployer/contract preset/a146eef9-b442-42f6-b31c-c19bfc1ee593.json [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      }
+    },
+    "es6-node": {
+      "cleanupFunction": "",
+      "dependencies": {},
+      "initFunction": "",
+      "inputFunctions": {
+        "543f8ad2-7755-44f6-bc9b-e6208c87b15a": "state.data = data;\noutputs.written(null, tag);",
+        "0b28df10-b056-4666-ae14-2357d747f7b9": "const value = state.data !== undefined \n  ? state.data\n  : params.data;\nif (value !== undefined)\n  outputs.data(value, tag);\nelse\n  outputs[\"not found\"](null, tag);",
+        "5c2df9e2-9902-488b-9381-5474fb89b84b": "//  number-like keys are not allowed since they\n/// do not preserve the strict key order\nif (Object.keys(data).some((x)=>!isNaN(x))){\n  throw new Error(\"number-like key names are not allowed\");\n}\n\nconst initArray = Object.values(data).map((value)=>`${value}`);\noutputs[\"init parameters array\"](initArray, tag);\n"
```
Comment (lvass74) [inline blockchain/ethereum/contracts/erc721/presets/erc721presetminterpauserautoid deployer/contract preset/a146eef9-b442-42f6-b31c-c19bfc1ee593.json:60]: Converting a dictionary to an array using Object.values does not guarantee the order of the elements in the output array. Values will get into the array by order of creation that is up to the user thus non deterministic.

```
["name", "symbol", "baseTokenURI"].map((key) => data[key].toString())
```

**Review by lvass74 — CHANGES_REQUESTED** [review]

File: blockchain/ethereum/contracts/compile and deploy contract preset/f4f89c90-1285-4803-b3c8-530254056164.json [hunk 2]
```diff
@@ -0,0 +1,204 @@
+{
+  "name": "blockchain/ethereum/contracts/compile and deploy contract preset",
```
Comment (csabee1979) [inline blockchain/ethereum/contracts/compile and deploy contract preset/f4f89c90-1285-4803-b3c8-530254056164.json:2]: Protoype name should be start uppercase: Compile and deploy contract preset

File: blockchain/ethereum/contracts/compile and deploy contract preset/f4f89c90-1285-4803-b3c8-530254056164.json [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+      "11c0a845-1293-4670-9390-0a0586122a6f": {
+        "type": "typeof `connect credentials` of `deploy contract`",
+        "name": "credentials"
+      }
+    },
+    "outputs": {
+      "2acecfad-46ec-48f8-80ea-332931523eef": {
+        "type": "typeof `events` of `deploy contract`",
+        "name": "events",
+        "description": "[Inherited from port `events` of `contract deployer`] \nMultiplexed events. Useful for monitoring progress.\nUse Demultiplexer to separate them."
```
Comment (csabee1979) [inline blockchain/ethereum/contracts/compile and deploy contract preset/f4f89c90-1285-4803-b3c8-530254056164.json:29]: On resusable prototye I would avoid the automatically generated [Inherited ...] text.

File: blockchain/ethereum/contracts/compile and deploy contract preset/f4f89c90-1285-4803-b3c8-530254056164.json [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+      },
+      "41ca2dbd-2701-45d8-8411-8dea422e9563": {
+        "type": "string",
+        "name": "source"
+      },
+      "147727db-184c-41df-90e3-e4ab656781a2": {
+        "type": "typeof `compile settings` of `split contract preset`",
+        "name": "compile settings",
+        "description": "[Inherited from port `compile settings` of `splitter B`] \nSends input data split into individual fields."
```
Comment (csabee1979) [inline blockchain/ethereum/contracts/compile and deploy contract preset/f4f89c90-1285-4803-b3c8-530254056164.json:38]: On resusable prototye I would avoid the automatically generated [Inherited ...] text.

File: blockchain/ethereum/contracts/compile preset/bf676a5e-7de7-47bb-9a9e-d9b15bedebe3.json [hunk 5]
```diff
@@ -0,0 +1,159 @@
+{
+  "name": "blockchain/ethereum/contracts/compile preset",
```
Comment (csabee1979) [inline blockchain/ethereum/contracts/compile preset/bf676a5e-7de7-47bb-9a9e-d9b15bedebe3.json:2]: Name should start uppercase.

File: blockchain/ethereum/contracts/compile preset/bf676a5e-7de7-47bb-9a9e-d9b15bedebe3.json [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+      "2f58e67b-36ae-4452-b0d3-74858ac029ad": {
+        "type": "typeof `compile settings` of `flat compiler`",
+        "name": "compile settings"
+      }
+    },
+    "outputs": {
+      "b20e6376-18be-4b40-9f7c-9dfcc5382497": {
+        "type": "typeof `value` of `item getter`",
+        "name": "compiled contract",
+        "description": "[Inherited from port `value` of `item getter`] \nIf found, sends the value corresponding to the specified key.\n\nExample:\n1"
```
Comment (csabee1979) [inline blockchain/ethereum/contracts/compile preset/bf676a5e-7de7-47bb-9a9e-d9b15bedebe3.json:28]: I would remove the [Inherited...] part.

File: blockchain/ethereum/contracts/contract preset/c791825d-ced0-4354-b066-d8488511dd4b.json [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+      "cleanupFunction": "",
+      "dependencies": {},
+      "initFunction": "",
+      "inputFunctions": {
+        "543f8ad2-7755-44f6-bc9b-e6208c87b15a": "state.data = data;\noutputs.written(null, tag);",
+        "0b28df10-b056-4666-ae14-2357d747f7b9": "const value = state.data !== undefined \n  ? state.data\n  : params.data;\nif (value !== undefined)\n  outputs.data(value, tag);\nelse\n  outputs[\"not found\"](null, tag);"
+      }
+    }
+  },
+  "name": "blockchain/ethereum/contracts/contract preset",
```
Comment (csabee1979) [inline blockchain/ethereum/contracts/contract preset/c791825d-ced0-4354-b066-d8488511dd4b.json:54]: Name should start uppercase.

File: blockchain/ethereum/contracts/erc20/internal/convert supply to 18 digits/e9b1083e-70b6-420a-97a7-a693670a8f0e.json [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+      }
+    },
+    "outputs": {
+      "0ceedb30-3431-403e-a4fd-e4d534fe457a": {
+        "type": "any",
+        "name": "out"
+      }
+    }
+  },
+  "name": "blockchain/ethereum/contracts/erc20/internal/convert supply to 18 digits",
```
Comment (csabee1979) [inline blockchain/ethereum/contracts/erc20/internal/convert supply to 18 digits/e9b1083e-70b6-420a-97a7-a693670a8f0e.json:108]: Name should start uppercase.

File: blockchain/ethereum/contracts/flat compiler/0823d223-40ba-4e1a-b408-91f620a4b543.json [hunk 9]
```diff
@@ -0,0 +1,95 @@
+{
+  "name": "blockchain/ethereum/contracts/flat compiler",
```
Comment (csabee1979) [inline blockchain/ethereum/contracts/flat compiler/0823d223-40ba-4e1a-b408-91f620a4b543.json:2]: Name should start uppercase.

**Review by csabee1979 — APPROVED** [review]

**Pull request merged**